bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/api/DraytonWiserApi.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    public void setSmartPlugManualMode(final int id, final boolean manualMode) throws DraytonWiserApiException {
+        final String payload = "{\"Mode\":\"" + (manualMode ? "Manual" : "Auto") + "\"}";
+        sendMessageToHeatHub(SMARTPLUG_ENDPOINT + id, "PATCH", payload);
+    }
+
+    public void setSmartPlugOutputState(final int id, final boolean outputState) throws DraytonWiserApiException {
+        final String payload = "{\"RequestOutput\":\"" + (outputState ? "On" : "Off") + "\"}";
+        sendMessageToHeatHub(SMARTPLUG_ENDPOINT + id, "PATCH", payload);
+    }
+
+    public void setSmartPlugAwayAction(final int id, final boolean awayAction) throws DraytonWiserApiException {
+        final String payload = "{\"AwayAction\":\"" + (awayAction ? "Off" : "NoChange") + "\"}";
+        sendMessageToHeatHub(SMARTPLUG_ENDPOINT + id, "PATCH", payload);
+    }
+
+    private synchronized @Nullable ContentResponse sendMessageToHeatHub(final String path, final HttpMethod method)
+            throws DraytonWiserApiException {
+        return sendMessageToHeatHub(path, method.asString(), "");
+    }
+
+    private synchronized @Nullable ContentResponse sendMessageToHeatHub(final String path, final String method,
+            final String content) throws DraytonWiserApiException {
+        // we need to keep track of the number of times that the heat hub has "failed" to respond.
+        // we only actually report a failure if we hit an error state 3 or more times
+        try {
+            logger.debug("Sending message to heathub: {}", path);
+            final StringContentProvider contentProvider = new StringContentProvider(content);
+            final ContentResponse response = httpClient
+                    .newRequest("http://" + configuration.networkAddress + "/" + path).method(method)
+                    .header("SECRET", configuration.secret).content(contentProvider).timeout(10, TimeUnit.SECONDS)
+                    .send();
+
+            if (logger.isTraceEnabled()) {
+                logger.trace("Reponse (Status:{}): {}", response.getStatus(), response.getContentAsString());
+            }
+            if (response.getStatus() == HttpStatus.OK_200) {
+                failCount = 0;
+                return response;
+            } else if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                failCount++;
+                if (failCount > 2) {
+                    throw new DraytonWiserApiException("Invalid authorization token");
+                }
+            } else {
+                failCount++;
+                if (failCount > 2) {
+                    throw new DraytonWiserApiException("Heathub didn't repond after " + failCount + " retries");
+                }
+            }
+        } catch (final TimeoutException e) {
+            failCount++;
+            if (failCount > 2) {
+                logger.debug("Heathub didn't repond in time: {}", e.getMessage(), e);
```

Comment:
It seems a bit unreasonable to log the stack trace when the network fails.
```suggestion
                logger.debug("Heathub didn't repond in time: {}", e.getMessage());
```

Reply:
Resolved in af8d624